### PR TITLE
PIMS-1886 and PIMS-1964: Run tests in the CI/CD pipeline

### PR DESCRIPTION
**Pull request by asanchezr**
## Changes

- Run `dotnet test` within the Jenkins pipeline and report any failures back
- Run `npm test` within the Jenkins pipeline and report any failures back
- Collect code coverage from frontend and backend tests (not yet reporting back to SonarQube)
- Updating documentation, including how to add & configure Jenkins Slaves
- Add `CI` env var to frontend builder image to avoid installing `husky` on it
- Remove unnecessary `test.yaml` templates - testing is done within the pipeline now 

#### jenkins-slave-npm

- Provides a docker image of the Nodejs 10 runtime for use as a Jenkins slave
- Includes `sonar-scanner-cli`

#### jenkins-slave-dotnet

- Provides a docker image of the .NET Core 3.1 runtime for use as a Jenkins slave
- Includes `coverlet` and `sonnar-runner` as global tools

#### jenkins-slave-zap

- Provides a docker image of the zap runtime for use as a Jenkins slave
- This is in preparation of future devops work to run ZAP security scans on the codebase

### CODE REVIEW

File: openshift/pipelines/Jenkinsfile.cicd [hunk 1]
```diff
@@ -210,47 +208,157 @@ pipeline {
         }
       }
     }
-    stage("Run App Tests") {
-      agent any
+    stage("FRONTEND Tests") {
+      agent {
+        label 'jenkins-slave-npm'
```
Comment (Fosol) [inline openshift/pipelines/Jenkinsfile.cicd:213]: I'm curious how this works?

File: openshift/templates/jenkins-slaves/jenkins-slave-generic-template.yaml [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+    value: 4000m
+  - name: MEMORY_REQUEST
+    description: Minimal amount of memory needed to run
+    displayName: Memory Request
+    value: 512Mi
+  - name: MEMORY_LIMIT
+    description: Maximum amount of memory allowed to use
+    displayName: Memory Limit
+    value: 4Gi
+labels:
```
Comment (Fosol) [inline openshift/templates/jenkins-slaves/jenkins-slave-generic-template.yaml:148]: Odd to put the labels at the bottom

Comment (asanchezr) [reply openshift/templates/jenkins-slaves/jenkins-slave-generic-template.yaml:148]: I copied this file as-is from a Red Hat repo. I'm fine to move them up.

File: openshift/pipelines/Jenkinsfile.cicd (hunk 1 above)
Comment (asanchezr) [reply openshift/pipelines/Jenkinsfile.cicd:213]: Yeah, it took a while to learn. Quite the learn curve. I can go over details on our next Dev Training session

**Pull request merged**